L1Trigger/Phase2DAQProducer/interface/DTCUnit.h

```diff
@@ -0,0 +1,217 @@
+#ifndef DTCUNIT_H
+#define DTCUNIT_H
+
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+
+#include "Phase2RawToDigi/Phase2DAQProducer/interface/Cluster.h"
+#include <fstream>
+
+class DTCUnit
+{
+    public:
+
+        enum DTCType { PS, S2, Unknown };
+
+        // initialize SLinkCollection with 4 FEDRawData objects
+        DTCUnit(unsigned int& event) : ClusterCollection(4, std::vector<std::vector<Cluster>>(18)), SLinkCollection(4), Type(DTCType::Unknown), EventID(event) 
+        {
+            
+        }
+
+        void insertHexWordAt(unsigned char *data_ptr, size_t word_index, uint32_t hex_word) 
+        {
+            data_ptr[word_index * 4 + 0] = (hex_word >> 24) & 0xFF;  // Most significant byte (bits 31-24)
```

Comment:
Magic numbers in the code are not allowed in CMSSW. Can we at very least define constants somewhere?

Reply:
ok. will update.

Reply:
done. look at Phase2DAQFormatSpecification.h and Phase2TrackerSpecifications.h under interface/.